pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -496,4 +518,16 @@ def load_raw_data(path):
             size = min(len(raw_data), len(timestamps))
             raw_data = raw_data[:size]
             timestamps = timestamps[:size]
-        yield from zip(raw_data, timestamps)
+
+        conf_data = load_worn_data(find_worn_path(timestamps_path))
+        if conf_data and len(conf_data) != len(timestamps):
+            logger.warning(
+                f"There is a mismatch between the number of confidence data ({len(conf_data)}) "
+                f"and the number of timestamps ({len(timestamps)})! Not using confidence data."
+            )
+            conf_data = None
+
+        if not conf_data:
+            conf_data = (1.0 for _ in range(len(timestamps)))
```

Comment:
Initially, I was incline to suggest using `np.ones_like(timestamps)` but given that this will be used in a `yield from` statement, I think your solution is prefarable.